js/app.js

```diff
@@ -11,29 +11,137 @@ Calculator.prototype.getHistoryAsString = function() {
+    }
+
+    let result = num1 - num2;
+    this.history.push(`${num1} - ${num2} = ${result}`);
+    alert(`${num1} - ${num2} = ${result}`);
+}
+
+Calculator.prototype.multiply = function(num1, num2) {
+    num1 = parseFloat(num1);
+    num2 = parseFloat(num2);
+
+    if (isNaN(num1) || isNaN(num2)) {
+        alert('Podane liczby są nieprawidłowe!');
+        return;
+    }
+
+    let result = num1 * num2;
+    this.history.push(`${num1} * ${num2} = ${result}`);
+    alert(`${num1} * ${num2} = ${result}`);
+}
+
+Calculator.prototype.divide = function(num1, num2) {
+    num1 = parseFloat(num1);
+    num2 = parseFloat(num2);
+
+    if (isNaN(num1) || isNaN(num2)) {
+        alert('Podane liczby są nieprawidłowe!');
+        return;
+    }
+
+    if (num2 === 0) {
+        alert('Nie można dzielić przez zero!');
+        return;
+    }
+
+    let result = num1 / num2;
+    this.history.push(`${num1} / ${num2} = ${result}`);
+    alert(`${num1} / ${num2} = ${result}`);
+}
+
+Calculator.prototype.power = function(base, exponent) {
+    base = parseFloat(base);
+    exponent = parseInt(exponent);
+
+    if (isNaN(base) || isNaN(exponent)) {
+        alert('Podane liczby są nieprawidłowe!');
+        return;
+    }
+
+    let result = 1;
+    for (let i = 0; i < Math.abs(exponent); i++) {
+        result *= base;
+    }
+
+    if (exponent < 0) {
+        result = 1 / result;
+    }
+
+    this.history.push(`${base} ^ ${exponent} = ${result}`);
+    alert(`${base} ^ ${exponent} = ${result}`);
 }
 
+
+
+
+
+
 const calc = new Calculator();
 let action, promptContent, isCorrectAction, number1, number2;
-do { 
-    promptContent = 'Podaj jaką operację chcesz wykonać (+, -, *, /, ^) i potwierdź. \n'; // \n - znak nowej linii
-    promptContent += 'Jeśli chcesz zrezygnować wciśnij Anuluj. \n';
+
+do {
+    promptContent = 'Podaj jaką operację chcesz wykonać (+, -, *, /, ^) i potwierdź. \n'; 
+    promptContent += 'Jeśli chcesz zrezygnować, wciśnij Anuluj. \n';
     promptContent += 'Lista poprzednich operacji: \n' + calc.getHistoryAsString();
 
     action = prompt(promptContent);
     isCorrectAction = calc.isCorrectAction(action);
-    if(isCorrectAction) {
+    
+    if (isCorrectAction) {
         number1 = prompt('Podaj liczbę nr 1');
         number2 = prompt('Podaj liczbę nr 2');
 
-        if(action === '+') {
-            calc.add(number1, number2);
+        switch(action) {
+            case '+':
+                calc.add(number1, number2);
+                break;
+            case '-':
+                calc.subtract(number1, number2);
+                break;
+            case '*':
+                calc.multiply(number1, number2);
+                break;
+            case '/':
+                calc.divide(number1, number2);
+                break;
+            case '^':
+                calc.power(number1, number2);
+                break;
+            default:
+                alert('Nieprawidłowa operacja');
+                break;
```

Comment:
Switch to taki ładniejszy if, ale warunek to warunek.
Można spróbować zmniejszyć ilość warunków przez coś takiego:
https://devmentor.pl/b/mniej-instrukcji-warunkowych